miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditRunController.java

```diff
@@ -190,9 +204,13 @@ public ModelAndView setupForm(Run run, PlatformType platformType, ModelMap model
       if (run.getId() == Run.UNSAVED_ID) {
         model.put("title", "New Run");
         model.put("multiplexed", false);
+        model.put("metrics", "[]");
       } else {
         model.put("title", "Run " + run.getId());
         model.put("multiplexed", isMultiplexed(run));
+        model.put("metrics",
+            getSources().map(source -> source.fetchMetrics(run)).filter(metrics -> !LimsUtils.isStringBlankOrNull(metrics))
```

Comment:
So Sonar understands streams, but not collectors... Or only knows about the collectors from the `Collectors` class and doesn't look at the interface maybe :slightly_frowning_face: